src/couch_log/src/couch_log_formatter.erl

```diff
@@ -467,12 +467,14 @@ format_meta(Meta) ->
     %% https://www.rfc-editor.org/rfc/rfc5424.html#section-6.3
     %% iut="3" eventSource="Application" eventID="1011"
     string:join(
-        maps:fold(
-            fun(K, V, Acc) ->
-                [to_str(K, V) | Acc]
-            end,
-            [],
-            Meta
+        lists:sort(
```

Comment:
[This test as written](https://github.com/apache/couchdb/blob/main/src/couch_log/test/eunit/couch_log_formatter_test.erl#L35-L44) can only pass on both OTP 25 and 26 if we sort the keys here, and then expect them in the sorted order.

Reply:
I think we should fix the test. Here is one suggestion. Define a macro 

```
-define(assertOneOff(A, B),
    ?assertEqual(lists:any(fun(E) -> B == E end, A), B)
).
```

Then provide all combinations

```
?assertOneOff(
    [
        "[foo=123 baz=\"baz\" bar=\"barStr\"]",
        "[foo=123 bar=\"barStr\" baz=\"baz\"]",
        "[baz=\"baz\" foo=123 bar=\"barStr\"]",
        "[baz=\"baz\" bar=\"barStr\" foo=123]",
        "[bar=\"barStr\" foo=123 baz=\"baz\"]",
        "[bar=\"barStr\" baz=\"baz\" foo=123]"    
    ],
    lists:flatten(Entry#log_entry.msg)
),
```

Reply:
While I don't love this test, I also don't think making it more complicated really improves it.

IMO those failing tests pointed out actual bugs in our code, which failed to specify the key order of a formatted map, leaving it up to the OTP implementation to decide. Now that the (undocumented) order changed, I think it would be best to sort the keys ourselves, and not continue relying on how OTP decides to order formatted map keys, and possibly keep having to fix the tests going forward.

What are the arguments against determining the formatted key order once and for all?

Reply:
I had a concern that sorting the map elements would harm the performance. However given the fact that reports would be small we can have a compromise. Which is every time we introduce a new report type we would try to select shorter names for keys. 

I am not going to block this PR on the map order grounds.